tools/issue-labeler/src/IssueLabelerService/Answers/OpenAiAnswerService.cs

```diff
@@ -37,37 +37,69 @@ public async Task<AnswerOutput> AnswerQuery(IssuePayload issue, Dictionary<strin
             // Query + Filtering configurations
             string query = $"{issue.Title} {issue.Body}";
             int top = int.Parse(_config.SourceCount);
-            double scoreThreshold = double.Parse(_config.ScoreThreshold);
-            double solutionThreshold = double.Parse(_config.SolutionThreshold);
+            var scoreThreshold = double.Parse(_config.ScoreThreshold);
+            var solutionThreshold = double.Parse(_config.SolutionThreshold);
+
+            //Subquery generation configurations
+            int minSubqueries = 3;
+            int maxSubqueries = 7;
+            int charsPerSubquery = 30;
+            string subqueryCount = Math.Max(minSubqueries, Math.Min(maxSubqueries, query.Length / charsPerSubquery)).ToString();
+            var replacementSubqueriesPrompt = new Dictionary<string, string>
+            {
+                { "subqueryCount", subqueryCount },
+                { "query", query }
+            };
 
-            var issues = await _ragService.SearchIssuesAsync(issueIndexName, issueSemanticName, issueFieldName, query, top, scoreThreshold, labels);
-            var docs = await _ragService.SearchDocumentsAsync(documentIndexName, documentSemanticName, documentFieldName, query, top, scoreThreshold, labels);
- 
-            if (docs.Count == 0 && issues.Count == 0)
+            var subqueriesPrompt = AzureSdkIssueLabelerService.FormatTemplate(_config.SubqueriesGenerationPrompt, replacementSubqueriesPrompt, _logger);
+
+            //Step 1: Generate sub-queries
+            var subqueries = await _ragService.GenerateSubqueriesAsync(subqueriesPrompt, query, modelName);
+
+            //Step 2: Retrieve results for sub-queries
+            var uniqueIssues = new HashSet<Issue>(new IssueIdComparer());
+            var uniqueDocs = new HashSet<Document>(new DocumentUrlComparer());
+
+            await RetrieveAndAggregateSubqueryResults(
+                subqueries,
+                issueIndexName,
+                issueSemanticName,
+                issueFieldName,
+                documentIndexName,
+                documentSemanticName,
+                documentFieldName,
+                top,
+                scoreThreshold,
+                labels,
+                uniqueIssues,
+                uniqueDocs
```

Comment:
Consider returning the unique issues and documents from RetrieveAndAggregateSubqueryResults instead of passing them as mutable sets. This would make the method signature cleaner.

```c#
private async Task<(HashSet<Issue>, HashSet<Document>)> RetrieveResultsAsync(
    IEnumerable<string> subqueries,
    int top,
    double threshold,
    Dictionary<string, string> labels)
```